### DISCUSSION TITLE
Frontend: Session nutzen mit Redaxo-Bordmitteln

### DISCUSSION THREAD

**Discussion by f5alex** [FriendsOfREDAXO/.github#10] [Q&A] [answered]
Kann man mit Redaxo-Bordmitteln im Frontend eigene Werte in einer Session speichern und lesen?
Oder: Welche Möglichkeiten gibts es um dem Besucher Inhalte nur einmal pro Besuch zu zeigen.

**Comment by marcohanke** [comment, marked as answer]
Kein Thema. In Redaxo werden alle Superglobals über rex_request abgehandelt (https://redaxo.org/doku/main/requests)

Für PHP Sessions in Redaxo machst du das so:
Du startest zuerst eine neue Session mit `rex_login::startSession();`
Danach fragst du eine Variabel deiner Wahl innerhalb der Session ab mit `rex_session('MeineVariable')`
Ist diese nicht gesetzt führst du deinen Code aus.

Anschließend setzt du die Variable mit `rex_set_session('MeineVariable', 'MeinWert');`

Ruft der Besucher die Seite erneut auf ist die Variable schon gesetzt und der Code wird nicht mehr ausgeführt.
Das ganze passiert so lange, bis der Besucher sein Browser schließt und die Session damit beendet ist.

Willst du das für alle Seiten deiner Website macht das ganze direkt im Template am meisten Sinn.

Komplett würde dass so aussehen

```
//Beim ersten Aufruf der Website wird die Session gestartet. Sie bleibt so lange aktiv, bis der Browser beendet wird.

if (rex_session('MeineVariable')) {
    //Code der ausgeführt ist, wenn die Variable gesetzt ist (direkt NACH dem ersten Aufruf der Website bis zum beenden des Browsers
} else {
    //Code der beim ersten Aufruf der Website ausgeführt wird bis die Variable 'MeineVariable' gesetzt ist
}
//Variable wird gesetzt
rex_set_session('MeineVariable', 'MeinWert');
```
